c2corg_ui/static/js/auth/authservice.js

```diff
@@ -89,32 +89,67 @@ app.Authentication.prototype.isAuthenticated = function() {
 /**
  * Checks if the current user has rights to access/edit the document.
  * TODO: rights to personal documents only for current user.
+ * @param {Array<string>} users
+ * @param {?string} imageType
  * @return {boolean}
  * @export
  */
-app.Authentication.prototype.hasEditRights = function(users) {
+app.Authentication.prototype.hasEditRights = function(users, imageType) {
   // not logged -> return false
   if (!this.isAuthenticated()) {
     return false;
   }
-  var userid = this.userData.id;
+
   var roles = this.userData.roles;
   // moderator has rigths
   if (roles.indexOf('moderator') > -1) {
     return true;
   }
-  // we are checking for editing rights of an outing
-  if (users) {
-    for (var i = 0; i < users.length; i++) {
-      if (userid === users[i].id) {
-        return true;
-      }
-    }
-    return false;
-    // by default, user has rights to every doc
+
+  if (users.length) {
+    return this.hasEditRightsOuting_(users);
+  // 'default' is set when imageType is Null in the mako template
+  } else if (imageType !== 'default') {
+    return this.hasEditRightsImage_(imageType);
+
   } else {
+    // by default, user has rights to every doc
     return true;
```

Comment:
You don't need to have an ``else`` block for a default ``return`` statement.
You can simply write:
```
if (cond A) {
  return 'foo';
}
return 'bar';
```